### PR TITLE
Change iOS PEM searcher to use PemEncoding.TryFindUtf8

### PR DESCRIPTION
**Pull request by vcsjones**
`AppleCertificatePal.TryDecodePem` can use the UTF-8 decoder. This extends out `PemEnumerator` to work on UTF-8 inputs.

### CODE REVIEW

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/AppleCertificatePal.Pem.iOS.cs [hunk 1]
```diff
@@ -2,59 +2,48 @@
...
-using Internal.Cryptography;
 
 namespace System.Security.Cryptography.X509Certificates
 {
     internal sealed partial class AppleCertificatePal : ICertificatePal
     {
-        internal delegate bool DerCallback(ReadOnlySpan<byte> derData, X509ContentType contentType);
-
-        internal static bool TryDecodePem(ReadOnlySpan<byte> rawData, DerCallback derCallback)
+        internal static void TryDecodePem(ReadOnlySpan<byte> rawData, Func<ReadOnlySpan<byte>, X509ContentType, bool> derCallback)
```
Comment (vcsjones) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/AppleCertificatePal.Pem.iOS.cs:12]: The return value from this method was always ignored and seemed to have no purpose any more. So it became `void`.

File: src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/AppleCertificatePal.Pem.iOS.cs [hunk 2]
```diff
@@ -2,59 +2,48 @@
...
+using System.Buffers.Text;
 using System.Diagnostics;
-using System.Text;
-using Internal.Cryptography;
 
 namespace System.Security.Cryptography.X509Certificates
 {
     internal sealed partial class AppleCertificatePal : ICertificatePal
     {
-        internal delegate bool DerCallback(ReadOnlySpan<byte> derData, X509ContentType contentType);
```
Comment (vcsjones) [inline src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/AppleCertificatePal.Pem.iOS.cs:13]: We don't need a dedicated delegate anymore with `allows ref struct` on Func now, so use Func.

**Comment by bartonjs** [comment]
Since it's marked as Draft I just skimmed, but it looks good, conceptually.

**Comment by vcsjones** [comment]
Marked as draft while I waited for CI in case I needed to make changes since I don't have iOS tests working locally.

Failures from runtime-extra-platforms appear unrelated, so marked ready for review.

**Review by bartonjs — APPROVED** [review]

**Pull request merged**